dvc/dependency/repo.py

```diff
@@ -122,6 +122,9 @@ def download(self, to):
                 self.def_path, self.def_repo[self.PARAM_URL]
             )
 
-    def update(self):
-        with self._make_repo(rev_lock=None) as repo:
+    def update(self, rev=None):
+        if not rev:
+            rev = self.def_repo.get("rev")
```

Comment:
No need to do that. See `_make_rev`, where it will merge your args with def_repo anyways.